app/controllers/users_controller.rb

```diff
@@ -25,22 +25,17 @@ def update
   end
 
   def create
-    if passwords_dont_match
-      redisplay_new_form "Passwords do not match"
 
-    elsif incomplete_information
-      redisplay_new_form "Please fill in all fields"
+    @user = User.new(user_info)
 
-    elsif email_in_use
-      @user = User.new(user_info.except(:email))
-      flash[:info] = "E-Mail already in use"
-      render :new
+    if @user.valid?
```

Comment:
Looks good. Will add a new issue to move errors to the Flash so we can use our existing flash display area to handle showing errors.